frontend/src/components/form/api/__tests__/ApiMock.js

```diff
@@ -0,0 +1,87 @@
+function mockPromiseResolving (value) {
+  return new Promise((resolve, reject) => {
+    const timer = setTimeout(() => {
+      clearTimeout(timer)
+      resolve(value)
+    }, 100)
+  })
+}
+
+class MockStubbing {
+  constructor (fieldName, value) {
+    this._fieldName = fieldName
+    this._value = value
+  }
+
+  forFieldName (fieldName) {
+    this._fieldName = fieldName
+    return this
+  }
+
+  get fieldName () {
+    return this._fieldName
+  }
+
+  get value () {
+    return this._value
+  }
+}
+
+class ApiMockState {
+  constructor () {
+    this._get = jest.fn()
+    this._patch = jest.fn()
+  }
+
+  getMocks () {
+    return {
+      get: this._get,
+      patch: this._patch
+    }
+  }
+
+  get () {
+    const apiMock = this
+    return {
+      thenReturn (mockStubbing) {
+        if (!(mockStubbing instanceof MockStubbing)) {
+          throw new Error('apiMock must be instance of MockStubbing')
+        }
+        if (mockStubbing.fieldName === undefined || mockStubbing.value === undefined) {
+          throw new Error('fieldName and value must be defined')
+        }
+        apiMock._get.mockReturnValue({
+          [mockStubbing.fieldName]: mockStubbing.value,
+          _meta: {
+            load: Promise.resolve(mockStubbing.value)
+          }
+        })
+      }
+    }
+  }
+
+  patch () {
+    const apiMock = this
+    return {
+      thenReturn (mockStubbing) {
+        if (!(mockStubbing instanceof MockStubbing)) {
+          throw new Error('apiMock must be instance of MockStubbing')
+        }
+        if (mockStubbing.fieldName !== undefined || mockStubbing.value === undefined) {
+          throw new Error('fieldName must be undefined and value must be defined')
+        }
+        apiMock._patch.mockReturnValue(mockPromiseResolving(mockStubbing.value))
+      }
+    }
+  }
+}
+
+export class ApiMock {
+  static create () {
+    return new ApiMockState()
+  }
+
+  static success (value) {
+    return new MockStubbing(undefined, value)
+  }
+}
```

Comment:
Wow, that's a lot of code for adding just a single test. This looks like something we could integrate directly in hal-json-vuex, would you agree? I am worried that this will be hard to maintain if it's sitting here.

Reply:
I started here and thought i don't want this massive object in the test just to mock the api.
For the next tests, my live will be easier.

It would be nice if hal-json-vuex would offer a way to mock itself in tests.
I would wait a little to look if it really helps to write shorter tests with less boilerplate, and then we can migrate it to hal-json-vuex.
I also did not check if my mocks represent the behaviour of hal-json-vuex correctly, it worked for that test.

With JUnit you normally just write the hierarchy in the test like:
const api = jest.fn()
const get = jest.fn()
api.mockReturnValue('get', get)
get.mockReturnValue

Maybe there is a reason to not write such a Mocking Helper, but to repeat yourself?